Krnlmon support for linux networking version 2

n-sandeep opened the pull request:
Support to add bridge_id key for FDB tables
Remove other p4 tables which are not used in this LNW version

ffoulkes commented on switchapi/es2k/switch_fdb.c:
```diff
@@ -291,9 +291,19 @@ switch_status_t switch_api_l2_forward_create(const switch_device_t device,
       CHECK_RET(status != SWITCH_STATUS_SUCCESS, status);
+
+      /* Get appropiate port_id if not already set */
+      switch_pd_to_get_port_id(&rif_info->api_rif_info);
```
`switch_pd_to_get_port_id()` has two failure cases. Shouldn't it return a status code?

n-sandeep replied:
in switch_pd_to_get_port_id(), error when returning "rmac info" not found or "rmac_entry" not found if fine. There are cases where valid rmac wouldnt have been created when we call this API.

ffoulkes commented on switchapi/es2k/switch_pd_utils.c:
```diff
@@ -194,6 +187,187 @@ tdi_status_t switch_pd_get_physical_port_id(switch_device_t device,
+
+  krnlmon_log_debug("Entered: %s", __func__);
+
+  status = tdi_info_get(dev_id, PROGRAM_NAME, &info_hdl);
```
These functions return the pointer to a handle, rather than the handle itself?

ffoulkes reviewed (approved):
LGTM
I posed one question and suggested that a function should return error status.

nupuruttarwar commented on switchapi/es2k/switch_fdb.c:
```diff
@@ -291,9 +291,19 @@ switch_status_t switch_api_l2_forward_create(const switch_device_t device,
+      /* Get bridge ID from its physical port ID */
+      status = switch_pd_get_bridge_id(device, api_l2_info->port_id,
+                                       &api_l2_info->bridge_id);
+      CHECK_RET(status != SWITCH_STATUS_SUCCESS, status);
```
What happens if the port is not part of any bridge?

n-sandeep replied:
It is an error case, we always need to associate a port to a bridge in LNW-v2

nupuruttarwar commented on switchapi/es2k/switch_pd_utils.c:
```diff
@@ -194,6 +187,187 @@ tdi_status_t switch_pd_get_physical_port_id(switch_device_t device,
+                      ACTION_L2_FWD_AND_BYPASS_BRIDGE_PARAM_PORT, status);
+    goto dealloc_resources;
+  }
+  //    status = tdi_data_field_get_value_ptr(data_hdl, data_field_id,
```
Can we remove this instead of commenting?

nupuruttarwar commented on switchapi/es2k/switch_pd_utils.c:
```diff
@@ -194,6 +187,187 @@ tdi_status_t switch_pd_get_physical_port_id(switch_device_t device,
+  const tdi_table_hdl* table_hdl = NULL;
+  const tdi_table_info_hdl* table_info_hdl = NULL;
+  uint32_t network_byte_order = 0;
+  uint64_t get_bridge_id = 0;
```
bridge_id is uint8_t. Do we really need uint64_t here?

n-sandeep replied:
tdi_data_field_get_value is taking uint64 as an argument which return the bridge value.
I can take it up as backlog and UT the changes to check if "tdi_data_field_get_value_ptr" can be used .

nupuruttarwar reviewed (commented):
Can we add more information to commit message descibinmg how lnw-version 2 is different than lnw-ver1? Also, please specify that with new version, IPv6 support is removed

vsureshkumarp commented on switchlink/switchlink_link.c:
```diff
@@ -88,6 +93,51 @@ static switchlink_link_type_t get_link_type(const char* info_kind) {
+
+  strncpy(drvname, drv.driver, sizeof(drvname));
+
+  if (!memcmp(drvname, "openvswitch", strlen(drvname)) ||
```
As part of improvements, create new macros for these driver names and utilize them when comparing drivers.

vsureshkumarp commented on switchlink/switchlink_link.c:
```diff
@@ -88,6 +93,51 @@ static switchlink_link_type_t get_link_type(const char* info_kind) {
+
+  r = ioctl(fd, SIOCETHTOOL, &ifr);
+  if (r) {
+    goto end;
```
It is always beneficial for debugging to log the error details at this point.  This applies to every instance in this function where error messages aren't logged before returning.

vsureshkumarp commented on switchapi/es2k/switch_pd_utils.c:
```diff
@@ -194,6 +187,184 @@ tdi_status_t switch_pd_get_physical_port_id(switch_device_t device,
+  uint32_t network_byte_order = 0;
+  uint64_t get_bridge_id = 0;
+
+  krnlmon_log_debug("Entered: %s", __func__);
```
Remove this log.

vsureshkumarp reviewed (approved):
LGTM

n-sandeep commented:
> * I posed one question and suggested that a function should return error status.
> * Have you done a test build to make sure dpdk isn't broken?
> * Also, you'll want to run clang-format on any source files you're committing.
>   `clang-format -i <file>...` will update them in place.

DPDK build went fine, addressed your comments and also ran clang-format
